src/Test/QuickCheck/Arbitrary.purs

```diff
@@ -223,3 +230,31 @@ genericArbitrary = to <$> (arbitrary :: Gen rep)
 genericCoarbitrary :: forall a rep. Generic a rep => Coarbitrary rep => a -> Gen a -> Gen a
 genericCoarbitrary x g = to <$> coarbitrary (from x) (from <$> g)
 
+-- | A helper typeclass to implement `Arbitrary` for records.
+class ArbitraryRowList
+        (list :: RowList)
+        (row :: # Type)
+        | list -> row where
+  arbitraryRecord :: RLProxy list -> Gen (Record row)
+
+instance arbitraryRowListNil :: ArbitraryRowList Nil () where
+  arbitraryRecord _ = pure {}
+
+instance arbitraryRowListCons ::
+  ( Arbitrary a
+  , ArbitraryRowList listRest rowRest
+  , RowLacks key rowRest
+  , RowCons key a rowRest rowFull
+  , RowToList rowFull (Cons key a listRest)
+  , IsSymbol key
+  ) => ArbitraryRowList (Cons key a listRest) rowFull where
+  arbitraryRecord _ = do
+     value <- arbitrary
+     previous <- arbitraryRecord (RLProxy :: RLProxy listRest)
+     pure $ insert (SProxy :: SProxy key) value previous
```

Comment:
This makes me think we should add `ST` support to `purescript-records`, so we can build the record in place.

Reply:
That said, I'm not sure what type we'd give to in-place _insertion_ of a new field :/